### DISCUSSION TITLE
Xiangqi Standard for FEN?

### DISCUSSION THREAD

**Discussion by nguyenpham** [fairy-stockfish/Fairy-Stockfish#544] [General]
I didn’t find any discussion about Xiangqi Standard on FEN strings thus I start a discussion here.

From my finding, there is an ambiguity in names and abbreviates for Xiangqi pieces. We have two main sets used for software:

**Pieces without ambiguity:**
King K (Wiki uses General G)
Advisor A 
Rook R
Cannon C
Pawn P (Wiki uses Soldier S)

**Pieces with ambiguities (1 vs 2)**
Horse H vs Knight N
Elephant E vs Bishop B

(1) used for Wikipedia (https://en.wikipedia.org/wiki/Xiangqi), some books, WinBoard, and some of its belonging Xiangqi engines 
(2) used mainly for all engines supported UCCI (Gaga, Sachess), some websites ([chessdb.cn](http://chessdb.cn/)) as well as Fairly-Stockfish 

What is your idea about those ambiguities? How to solve the problem? Should we do something or ignore it completely?

Thus note that at the moment even Fairly-Stockfish is using (2).

**Comment by ianfab** [comment]
The UCCI and UCI-cyclone protocol to my knowledge use NB, for UCCI it can be seen from the spec as well https://www.xqbase.com/protocol/cchess_ucci.htm. NB are basically never used for things that are visible to users, because HE is the standard in xiangqi. Therefore the only place where I have seen NB is in FENs. However, in the xboard protocol HE seems more common even in FEN. Fairy-Stockfish simply allows both on input side in all protocols to be on the safe side. Generally my recommendation would be to use NB for UCI, UCI-cyclone, and UCCI, but to use HE for CECP.

That Fairy-SF uses 1 based rank counting in UCCI is not true. I follow the above spec as well as https://fairy-stockfish.github.io/chess-variant-standards/.

**Reply by nguyenpham** [reply]
Sorry, you are right. Just remove the sentence about Fairy-SF using a rank counter different from UCCI.
